src/App.tsx

```diff
@@ -1,204 +1,189 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { client } from './utils/fetchClient';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+
+interface Todo {
+  id: number;
+  userId: number;
+  title: string;
+  completed: boolean;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [loadingId, setLoadingId] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const timeoutRef = useRef<number | null>(null);
+
+  const filteredTodos = useMemo<Todo[]>(() => {
+    if (filter === 'all') {
+      return todos;
+    }
+
+    if (filter === 'active') {
+      return todos.filter(t => !t.completed);
+    }
+
+    if (filter === 'completed') {
+      return todos.filter(t => t.completed);
+    }
+
+    return todos;
+  }, [todos, filter]);
+
+  function showError(message: string) {
```

Comment:
One method of creating functions must be followed, fix it everywhere
```suggestion
  const showError = (message: string) => {
```